soc/arm/nxp_imx/mimx8mm6_m4/soc.c

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2019, Manivannan Sadhasivam <mani@kernel.org>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <fsl_clock.h>
+#include <fsl_common.h>
+#include <fsl_rdc.h>
+#include <init.h>
+#include <kernel.h>
+#include <soc.h>
+
+#include <dt-bindings/rdc/imx_rdc.h>
+
+/* OSC/PLL is already initialized by ROM and Cortex-A7 (u-boot) */
+void SOC_RdcInit(void)
+{
+	/* Move M4 core to specific RDC domain 1 */
+	rdc_domain_assignment_t assignment = {0};
+
+	assignment.domainId = M4_DOMAIN_ID;
+	RDC_SetMasterDomainAssignment(RDC, kRDC_Master_M4, &assignment);
+
+	/*
+	 * The M4 core is running at domain 1, enable clock gate for
+	 * Iomux to run at domain 1.
+	 */
+	CLOCK_EnableClock(kCLOCK_Iomux0);
+	CLOCK_EnableClock(kCLOCK_Iomux1);
+	CLOCK_EnableClock(kCLOCK_Iomux2);
+	CLOCK_EnableClock(kCLOCK_Iomux3);
+	CLOCK_EnableClock(kCLOCK_Iomux4);
+
+	CLOCK_EnableClock(kCLOCK_Qspi);
+
+	/*
+	 * The M4 core is running at domain 1, enable the PLL clock sources
+	 * to domain 1.
+	 */
+	/* Enable SysPLL1 to Domain 1 */
+	CLOCK_ControlGate(kCLOCK_SysPll1Gate, kCLOCK_ClockNeededAll);
+	/* Enable SysPLL2 to Domain 1 */
+	CLOCK_ControlGate(kCLOCK_SysPll2Gate, kCLOCK_ClockNeededAll);
+	/* Enable SysPLL3 to Domain 1 */
+	CLOCK_ControlGate(kCLOCK_SysPll3Gate, kCLOCK_ClockNeededAll);
+	/* Enable AudioPLL1 to Domain 1 */
+	CLOCK_ControlGate(kCLOCK_AudioPll1Gate, kCLOCK_ClockNeededAll);
+	/* Enable AudioPLL2 to Domain 1 */
+	CLOCK_ControlGate(kCLOCK_AudioPll2Gate, kCLOCK_ClockNeededAll);
+	/* Enable VideoPLL1 to Domain 1 */
+	CLOCK_ControlGate(kCLOCK_VideoPll1Gate, kCLOCK_ClockNeededAll);
+}
+
+/* AUDIO PLL1 configuration */
+const ccm_analog_frac_pll_config_t g_audioPll1Config = {
+	.refSel  = kANALOG_PllRefOsc24M, /* PLL reference OSC24M */
+	.mainDiv = 655U,
+	.dsm     = 23593U,
+	.preDiv  = 5U,
+	.postDiv = 2U, /* AUDIO PLL1 frequency  = 786432000HZ */
+};
+
+/* AUDIO PLL2 configuration */
+const ccm_analog_frac_pll_config_t g_audioPll2Config = {
```

Comment:
make these configs and all the functions in this file `static`

Reply:
okay